# Generated by Django 2.1.3 on 2019-11-18 11:59

from django.db import migrations,models

def pull_up_names(apps, schema_editor):
    # We can't import the Person model directly as it may be a newer
    # version than this migration expects. We use the historical version.
    Series = apps.get_model('retconcreatives', 'Series')
    Episode = apps.get_model('retconcreatives', 'Episode')
    CW = apps.get_model('retconcreatives', 'CreativeWork')
    for item in Series.objects.all():
        c= CW.objects.get(id=item.creativework_ptr_id)
        c.tmpname=item.name
        c.save()
    for item in Episode.objects.all():
        c= CW.objects.get(id=item.creativework_ptr_id)
        c.tmpname=item.name
        c.save()

class Migration(migrations.Migration):

    dependencies = [
        ('retconcreatives', '0019_auto_20191118_1117'),
    ]

    operations = [

        migrations.AddField(
            model_name='creativework',
            name='tmpname',
            field=models.CharField(default='!Error:RetconError:MissingName!!!!!!!!!!!!!!!', max_length=64),
            preserve_default=False,
        ),
        
        migrations.RunPython(pull_up_names),
        migrations.RemoveField(
            model_name='episode',
            name='name',
        ),
        migrations.RemoveField(
            model_name='series',
            name='name',
        ),
        migrations.RenameField(
            model_name='creativework',
            old_name='tmpname',
            new_name='name'
        )

    ]
